src/sshagent/SSHAgent.cpp

```diff
@@ -449,7 +466,7 @@ void SSHAgent::databaseLocked(QSharedPointer<Database> db)
 
 void SSHAgent::databaseUnlocked(QSharedPointer<Database> db)
 {
-    if (!db || !isEnabled()) {
+    if (!db || !isEnabled() || (!useOpenSSH() && !usePageant())) {
```

Comment:
This wouldn't compile on any other platform than Windows. Though you could just remove this logic completely as we can return an error later, see previous comment.